modules/registry/registry.go

```diff
@@ -205,6 +203,47 @@ func (c *RegistryContainer) PushImage(ctx context.Context, ref string) error {
 	return c.ImageExists(ctx, ref)
 }
 
+// PullImage pulls an image from an external Registry.
+// It will use the internal registry to store the image.
+// This function is helpful when you want to push an image to your Registry
+// instance made by testcontainer.
+func (c *RegistryContainer) PullImage(ctx context.Context, ref string) error {
+	dockerCli, err := testcontainers.NewDockerClientWithOpts(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to create Docker client: %w", err)
```

Comment:
suggestion: describe the action instead that it was an failure as that makes wrapped messages easier to understand reducing unneeded content.

Move below.
```suggestion
		return fmt.Errorf("create docker client: %w", err)
```

Reply:
Not sure if there was an issue with the push, but not seeing this as resolved so marking as such.

Reply:
don't know why, I tought it was committed. going to add this manually.